IdentityCore/src/broker_operation/response/MSIDBrokerOperationGetAccountsResponse.m

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#import "MSIDBrokerOperationGetAccountsResponse.h"
+#import "MSIDAccount.h"
+
+@implementation MSIDBrokerOperationGetAccountsResponse
+
+#pragma mark - MSIDJsonSerializable
+
+- (instancetype)initWithJSONDictionary:(NSDictionary *)json error:(NSError **)error
+{
+    self = [super initWithJSONDictionary:json error:error];
+    
+    if (self)
+    {
+        if (![json msidAssertType:NSDictionary.class ofKey:@"response_data" required:YES error:error])
+        {
+            return nil;
+        }
+        NSDictionary *responseData = json[@"response_data"];
+        
+        if (![responseData msidAssertType:NSArray.class ofKey:@"accounts" required:YES error:error])
+        {
+            return nil;
+        }
+        NSArray *accountsJson = responseData[@"accounts"];
+        
+        NSMutableArray *accounts = [NSMutableArray new];
+        for (NSDictionary *accountJson in accountsJson)
+        {
+            MSIDAccount *account = [[MSIDAccount alloc] initWithJSONDictionary:accountJson error:error];
+            if (!account) return nil;
```

Comment:
should we fail entire operation if one of accounts fails to parse?

Reply:
After second thought, I think it might make more sense to return whatever accounts we could parse from the response, just in case some of the accounts are corrupted during data transition.

Now changed :)